routes/index.js

```diff
@@ -324,19 +322,26 @@ router.post('/updateaccount', common.isAuthenticated, (req, res, next) => {
         worldwide: req.body.worldwide,
         aboutMe: req.body.aboutMe
     }
-    sql.pool.query('UPDATE Users SET firstName=?, lastName=?, email=?, address=?, city=?, state=?, postalCode=?, country=?, worldwide=?, aboutMe=? WHERE id=?',
-    [data.firstName, data.lastName, data.email, data.address, data.city, data.state, data.postalCode, data.country, data.worldwide, data.aboutMe, data.userId],
+    sql.pool.query('SELECT id, email FROM Users WHERE email=?', [data.email],
     (err, results) => {
-        if (err) {
-            req.flash('error', err)
-            res.render('auth/updateaccount', data)
+        if (results.length != 0 && results[0].id != req.session.userId) {
+            req.flash("error", "This email is already associated with another account")
+            res.reload();  //this might need to be changed
```

Comment:
If the email already exists you won't want to do a reload as you'll clear out all the data they already entered. You should be using res.render here and passing back the data object you have. Typically in this case I would se the data.email to blank before sending back so it clears that field but leaves the rest. I'd also put back the title in the data so you're returning and still have the tab id or page title which should be using that as well.

Reply:
my new commit reflects these changes.